runbooks/koji/upgrade.md

```diff
@@ -0,0 +1,84 @@
+Koji upgrade
+==============
+
+Issue
+-----
+* You want to upgrade [Koji](http://koji.katello.org) to a newer version
+
+Resolution
+----------
+* Shut down koji services on `koji-fedora28-builder` and `kojibuilder1`.
+  * koji-fedora28-builder
+    ```sh
+    systemctl stop kojid
+    ```
+  * kojibuilder1
+	  ```sh
+    systemctl stop kojid
+	  systemctl stop kojira
+	  systemctl stop httpd
+	  systemctl stop postgresql
+    ```
+* Take snapshots in aws.
+  * EBS -> Volumes
+  * Right click on “koji 4 root volume” -> create snapshot
+  * Right click on “koji4 data volume /mnt/koji” -> create snapshot
+    * You can still use the host os while snapshot is `pending`,
+		it refers to the transfer to amazon s3.
+
+* update vms to latest software.
+  * kojibuilder1
+	  * `# yum update`
+	  * Remove `/etc/httpd/conf.d/{kojihub,kojiweb,ssl}.conf`
+      * We have our configs in `/etc/httpd/conf.modules.d` at the moment
+  * koji-fedora28-builder
+    ```sh
+    systemctl disable kojid  # to keep it from erroring out across reboots
+
+    dnf upgrade --refresh
+    
+    # if Fedora is out of date, upgrade it    
+    dnf install dnf-plugin-system-upgrade
+    # for each release, upgrade one at a time
+    dnf system-upgrade download --refresh --releasever={current + 1}
+    dnf system-upgrade reboot
+    ```
+
+* Run koji database migrations
+  ```sh
+  # for each .y step, run that .y's migration script
+  psql koji koji  </usr/share/doc/koji-{version}/docs/schema-upgrade-1.{y-1}-1.{y}.sql
+  ```
+* Start koji services on koji-fedora28-builder and kojibuilder1.
+  * kojibuilder1
+ 	  ```sh
+    systemctl start postgresql
+	  systemctl start httpd
+	  systemctl start kojira
+	  systemctl start kojid
+    ```
+  * koji-fedora28-builder
+    ```sh
+    systemctl start kojid
+    systemctl enable kojid
+    ```
+* Verify successful scratch build against a tag
```

Comment:
We found that actually tagging a build also uncovered some issues so I'd recommend building some nightly.